internal/testsacc/edgegw_nat_rule_resource_test.go

```diff
@@ -1,285 +1,318 @@
+				},
+				// ! Create testing
+				Create: testsacc.TFConfig{
+					TFConfig: testsacc.GenerateFromTemplate(resourceName, `
+					resource "cloudavenue_edgegateway_nat_rule" "example" {
+						edge_gateway_id = cloudavenue_edgegateway.example.id
+
+						name        = {{ generate . "name" }}
+						rule_type   = "SNAT"
+						description = {{ generate . "description" }}
+
+						# Using primary_ip from edge gateway
+						external_address         = "89.32.25.10"
+						internal_address         = "11.11.11.0/24"
+						snat_destination_address = "8.8.8.8"
+
+						priority = 10
+					}`),
+					Checks: []resource.TestCheckFunc{
+						// ! base checks
+						resource.TestCheckResourceAttr(resourceName, "name", testsacc.GetValueFromTemplate(resourceName, "name")),
+						resource.TestCheckResourceAttr(resourceName, "external_address", "89.32.25.10"),
+						resource.TestCheckResourceAttr(resourceName, "description", testsacc.GetValueFromTemplate(resourceName, "description")),
+						resource.TestCheckResourceAttr(resourceName, "internal_address", "11.11.11.0/24"),
+						resource.TestCheckResourceAttr(resourceName, "rule_type", "SNAT"),
+						resource.TestCheckResourceAttr(resourceName, "snat_destination_address", "8.8.8.8"),
+						resource.TestCheckResourceAttr(resourceName, "priority", "10"),
+					},
+				},
+				// ! Update testing
+				Updates: []testsacc.TFConfig{
+					{
+						TFConfig: testsacc.GenerateFromTemplate(resourceName, `
+						resource "cloudavenue_edgegateway_nat_rule" "example" {
+							edge_gateway_id = cloudavenue_edgegateway.example.id
+
+							name        = {{ get . "name" }}
+							rule_type   = "SNAT"
+							description = {{ generate . "description" }}
+
+							# Using primary_ip from edge gateway
+							external_address         = "89.32.25.10"
+							internal_address         = "11.11.11.0/24"
+							snat_destination_address = "9.9.9.9"
+
+							priority = 0
+						}`),
+						Checks: []resource.TestCheckFunc{
+							resource.TestCheckResourceAttr(resourceName, "name", testsacc.GetValueFromTemplate(resourceName, "name")),
+							resource.TestCheckResourceAttr(resourceName, "external_address", "89.32.25.10"),
+							resource.TestCheckResourceAttr(resourceName, "description", testsacc.GetValueFromTemplate(resourceName, "description")),
+							resource.TestCheckResourceAttr(resourceName, "internal_address", "11.11.11.0/24"),
+							resource.TestCheckResourceAttr(resourceName, "rule_type", "SNAT"),
+							resource.TestCheckResourceAttr(resourceName, "snat_destination_address", "9.9.9.9"),
+							resource.TestCheckResourceAttr(resourceName, "priority", "0"),
+							resource.TestCheckNoResourceAttr(resourceName, "dnat_external_port"),
+						},
+					},
+				},
+				// ! Imports testing
+				Imports: []testsacc.TFImport{
+					{
+						ImportState:       true,
+						ImportStateVerify: true,
+						ImportStateIDFunc: testAccNATRuleResourceImportStateIDFuncWithIDAndName(resourceName),
```

Comment:
This is unnecessary. The ImportStateIDBuilder builde automaticaly the ImportStateIDFunc